pkg/apis/fleet.cattle.io/v1alpha1/gitrepo_types.go

```diff
@@ -170,6 +170,9 @@ type GitRepoStatus struct {
 	// Commit is the Git commit hash from the last git job run.
 	// +nullable
 	Commit string `json:"commit,omitempty"`
+	// WebhookCommit is the latest Git commit hash received from a webhook
+	// +nullable
```

Comment:
In light of #2901, I wonder if we might want to make this field `+optional` instead of `+nullable`, to prevent it from appearing with an empty value on `GitRepo`s which rely solely on polling. WDYT?

Reply:
I think that being a commit hash... it makes sense to have the value as `+nullable` rather than being an empty string. 
So either we have a valid commit hash or we have nothing.

Reply:
> either we have a valid commit hash or we have nothing

Yes, precisely, if you meant to say that this field should either contain a valid value, or not appear at all; I think that this is what `+optional` achieves, not `+nullable`, as per [these docs](https://book.kubebuilder.io/reference/markers/crd-validation.html).

Reply:
oh, right, in that case I'll make this one and the normal commit both non `+nullable`